### DISCUSSION TITLE
GitHub Api

### DISCUSSION THREAD

**Discussion by Arman-ali-khan-786** [community/community#52656] [Apps, API and Webhooks]
How do it get the Api for GitHub to fetch all the details of a particular user of repo and based on other parameters...

**Comment by samus-aran** [comment]
Welcome to the GitHub Community, @Arman-ali-khan-786 , we're happy you're here! You are more likely to get a useful response if you are posting your question(s) in the applicable category and are explicit about what your project entails--giving a few more details might help someone give you a nudge in the right direction. I've gone ahead and moved it for you. Good luck!

**Comment by byrneh** [comment]
@Arman-ali-khan-786 

With the users login there are few APIs avaiable that can be called to retrueve the various information about a user
REST API [get-a-user](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-a-user)
REST API [get-contextual-information-for-a-user](https://docs.github.com/en/rest/users/users?apiVersion=2022-11-28#get-contextual-information-for-a-user)
GRAPHQL API [user](https://docs.github.com/en/graphql/reference/queries#user) the fields that can be retrieved are here [Fields for a User](https://docs.github.com/en/graphql/reference/objects#user)
The documentation also provides details on how to call and authenticate to each type of endpoint.

**Reply by Arman-ali-khan-786** [reply]
Thank you 😊
